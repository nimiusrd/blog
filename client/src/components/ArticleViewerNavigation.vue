<template>
    <div class="navigation">
        <button class="button edit" @click="handleClickEdit">
            Edit
        </button>
        <button class="button home" @click="handleClickHome">
            Home
        </button>
        <button class="button delete" @click="handleClickDelete">
            Delete
        </button>
    </div>
</template>

<script lang="ts">
import Vue from 'vue'

const ArticleViewerNavigation = Vue.extend({
    methods: {
        deleteArticle() {
            const id = this.$route.params.id

            fetch(
                `/api/articles/${id}`,
                {
                    method: 'DELETE'
                }
            )
                .catch(reason => {
                    console.log(reason)
                })
        },
        handleClickDelete() {
            this.deleteArticle()
            this.$router.push({
                path: '/'
            })
        },
        handleClickEdit() {
            const id = this.$route.params.id

            this.$router.push({
                path: `/articles/${id}/edit`
            })
        },
        handleClickHome() {
            this.$router.push({
                path: '/'
            })
        }
    }
})

export default ArticleViewerNavigation
</script>

<style scoped>
.button {
    font-size: 20px;
    font-weight: 500;
    margin: 8px;
    min-width: 100px;
    padding: 16px;
}

.navigation {
    align-items: center;
    justify-content: left;
    display: flex;
}
</style>
