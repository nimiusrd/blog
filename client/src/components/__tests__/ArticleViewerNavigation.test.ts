import ArticleViewerNavigation from '@/components/ArticleViewerNavigation.vue'
import {shallowMount} from '@vue/test-utils'

test('snapshot', () => {
    const wrapper = shallowMount(ArticleViewerNavigation)

    expect(wrapper.html()).toMatchSnapshot()
})
