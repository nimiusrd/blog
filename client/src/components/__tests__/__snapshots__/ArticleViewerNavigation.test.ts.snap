// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
"<div class=\\"navigation\\"><button class=\\"button edit\\">
        Edit
    </button> <button class=\\"button home\\">
        Home
    </button> <button class=\\"button delete\\">
        Delete
    </button></div>"
`;
